[UPG FAIL] (x4) elevate-cpanel:1787 "can" on undefined; elevate-cpanel:3848 command failed, fix it (cP:95050514)

Hi there,

We have 3x machines which have failed with the below:

```
* 2023-04-27 11:54:16 (3796) [ERROR] The elevation process failed during stage 4.
* 2023-04-27 11:54:16 (3796) [ERROR] 
* 2023-04-27 11:54:16 (3796) [ERROR] You can continue the process after fixing the errors by running:
* 2023-04-27 11:54:16 (3796) [ERROR] 
* 2023-04-27 11:54:16 (3796) [ERROR]     /usr/local/cpanel/scripts/elevate-cpanel --continue
* 2023-04-27 11:54:16 (3796) [ERROR] 
* 2023-04-27 11:54:16 (3796) [ERROR] You can check the error log by running:
* 2023-04-27 11:54:16 (3796) [ERROR] 
* 2023-04-27 11:54:16 (3796) [ERROR]     /usr/local/cpanel/scripts/elevate-cpanel
* 2023-04-27 11:54:16 (3796) [ERROR] 
* 2023-04-27 11:54:16 (3796) [ERROR] Last Error:
* 2023-04-27 11:54:16 (3796) [ERROR] 
* 2023-04-27 11:54:16 (3796) [ERROR] Can't call method "can" on an undefined value at /usr/local/cpanel/scripts/elevate-cpanel line 1787.
* 2023-04-27 11:54:17 (4891) [FATAL] Can't call method "can" on an undefined value at /usr/local/cpanel/scripts/elevate-cpanel line 1787.
```

And we have another 1x machine which has failed with:

```
* 2023-04-27 11:56:42 (3796) [ERROR] The elevation process failed during stage 4.
* 2023-04-27 11:56:42 (3796) [ERROR] 
* 2023-04-27 11:56:42 (3796) [ERROR] You can continue the process after fixing the errors by running:
* 2023-04-27 11:56:42 (3796) [ERROR] 
* 2023-04-27 11:56:42 (3796) [ERROR]     /usr/local/cpanel/scripts/elevate-cpanel --continue
* 2023-04-27 11:56:42 (3796) [ERROR] 
* 2023-04-27 11:56:42 (3796) [ERROR] You can check the error log by running:
* 2023-04-27 11:56:42 (3796) [ERROR] 
* 2023-04-27 11:56:42 (3796) [ERROR]     /usr/local/cpanel/scripts/elevate-cpanel
* 2023-04-27 11:56:42 (3796) [ERROR] 
* 2023-04-27 11:56:42 (3796) [ERROR] Last Error:
* 2023-04-27 11:56:42 (3796) [ERROR] 
* 2023-04-27 11:56:42 (3796) [ERROR] command failed. Fix it and run command. at /usr/local/cpanel/scripts/elevate-cpanel line 3848.
* 2023-04-27 11:56:44 (4891) [FATAL] command failed. Fix it and run command. at /usr/local/cpanel/scripts/elevate-cpanel line 3848.
```

With all 4 machines, we initially ran `--check` and cleared any blockers (packages/repos only), then elevated with `--start`

All machines had kernels updated and reboots applied before attempting the elevations.

Thanks,
Luke

--- a/lib/Elevate/Components/PerlXS.pm
+++ b/lib/Elevate/Components/PerlXS.pm
@@ -34,8 +34,8 @@ sub pre_leapp ($self) {
 
 sub post_leapp ($self) {
 
-    restore_perl_xs(DISTRO_PERL_XS_PATH);
-    restore_perl_xs( $Config{'installsitearch'} );
+    $self->restore_perl_xs(DISTRO_PERL_XS_PATH);
+    $self->restore_perl_xs( $Config{'installsitearch'} );
 
     return;
 }
@@ -109,7 +109,7 @@ sub purge_perl_xs ( $self, $path ) {
     return;
 }
 
-sub restore_perl_xs ($path) {
+sub restore_perl_xs ( $self, $path ) {
     my $stash = cpev::read_stage_file();
 
     if ( $path eq DISTRO_PERL_XS_PATH ) {
@@ -117,7 +117,7 @@ sub restore_perl_xs ($path) {
 
         if ( scalar keys %$rpms ) {    # If there are no XS modules to replace, there is no point to running the dnf install:
             my @cmd = ( '/usr/bin/dnf', '-y', '--enablerepo=epel', '--enablerepo=powertools', 'install', sort keys %$rpms );
-            __PACKAGE__->ssystem(@cmd);
+            $self->ssystem(@cmd);
         }
     }
 

